### PR TITLE
Groups List Page

### LINKED ISSUE #72: Create Groups List Page
Create Card component to display a user's groups.

### LINKED ISSUE #79: Add TypeScript to the project
https://create-react-app.dev/docs/adding-typescript/

### PR DESCRIPTION
**Pull request by Waidhoferj**
Fixes #72 
## Changes
**Primary change:** I added the Groups List Page page to the application.

**Other Stuff:**
- Added typescript to the application (Fixes #79). You can suffix your files in `.ts` or `.tsx` and you'll get a full static type checker.
- Added group metrics in order to track number of members and number of artworks.
- Added user/group assignments to the test server.
- Fixed overflow bug where scrolling pages would scroll past the nav bar.
- Added group icon to the nav bar.

## Testing
- Start frontend and backend servers.
- In a browser go to: [`http://localhost:3000/groups`](http://localhost:3000/groups)
- Ensure that the page populates with groups and scrolls correctly. Long pressing on a group should trigger a small animation but doesn't go anywhere.
- Turn off the backend server and ensure an error message is displayed.

Btw this really took 5 hrs. The bugs were surprisingly nasty.

### CODE REVIEW

File: frontend/src/pages/Groups/GroupsList/GroupsList.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    }
+  }
+`;
+
+export default function GroupsList() {
+  let groups = useGroupList();
+  if (groups.length) {
+    groups = new Array(20).fill(groups[0]);
+  }
```
Comment (Waidhoferj) [inline frontend/src/pages/Groups/GroupsList/GroupsList.jsx]: This is just for you all to try out a scrolling display. The code duplicates the first group 20x. I'll take it out before we merge.

**Review by gwholland3 — COMMENTED** [review]
Visually, everything looks great. However, at least on my end, every time I refresh the groups page it takes a good 1-1.5 seconds to actually load (and displays the "Doesn't look like you've joined any groups" message in the meantime), which I haven't noticed happening for other pages.

**Comment by Waidhoferj** [comment]
@gwholland3 Since we are loading several images per card, the load time is going to be a little longer. I added a `Spinner` component and cached the result of `useGroupList` with the `useMemo` hook. Lmk what you think.

**Review by Braden50 — APPROVED** [review]
All runs and looks good. Beautiful work!

I guess a small note is that I get quite a bit of warnings from the frontend including it saying my typescript version isn't the one I want exactly. All runs well though, and I trust you know what you're doing.

**Review by gwholland3 — APPROVED** [review]
Ooh very nice, the spinner looks great!

**Comment by Waidhoferj** [comment]
@Braden50 Most of those errors were there before. I'll correct them in another PR

**Comment by kyleshare** [comment]
Saw this was just closed but just ran it, looks great!

**Review by kyleshare — APPROVED** [review]

**Pull request merged**